game/match/match.go

```diff
@@ -2006,3 +2007,18 @@ func (m *Match) handleAdminMesseges(message string, user db.User) {
 
 	}
 }
+
+func (m *Match) handleMesseges(message string) {
+
+	currentPlayer := m.CurrentPlayer().Player
+	msgParts := strings.Split(message, " ")
+	if len(msgParts) < 1 {
+		return
+	}
+
+	switch msgParts[0] {
+	case "/quit":
+		m.End(m.Opponent(currentPlayer), fmt.Sprintf("%s won by opponent resigning.", m.Opponent(currentPlayer).Username()))
```

Comment:
Since incoming chat events are of ParallelEvent type they will be handled in a separate non-blocking goroutine than the match logic itself which might cause some panics during runtime as it can dispose of the match and players (which closes channels and nils most slices and pointers) at the same time as some match logic is executed which is trying to use those references

Maybe a separate Forfeit event of SequentialEvent type could be created and sent by the client, or some other way to make sure the m.End call is called from inside the blocking eventloop queue